lib/cloudant.js

```diff
@@ -75,7 +74,30 @@ Cloudant.prototype.getTypes = function() {
  */
 Cloudant.prototype.connect = function(cb) {
   debug('Cloudant.prototype.connect');
-  cb(null, this.cloudant);
+  var self = this;
+ // strip db name if defined in path of url before
+  // sending it to our driver
+  if (self.options.url) {
+    var parsedUrl = url.parse(self.options.url);
+    if (parsedUrl.path && parsedUrl.path !== '/') {
+      self.options.url = self.options.url.replace(parsedUrl.path, '');
+      if (!self.options.database)
+        self.options.database = parsedUrl.path.split('/')[1];
+    }
+  }
+  self.CloudantDriver(self.options, function(err, nano) {
+    if (err) return cb(err);
+    self.cloudant = nano;
+    if (self.options.database) {
+      // check if database exists
+      self.cloudant.db.get(self.options.database, function(err) {
```

Comment:
@b-admike do we have to check `database` exist? I think we can leave the server to return error if user calls an api against a non-existing database.